server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/jsieve/RejectAction.java

```diff
@@ -108,17 +114,26 @@ public void execute(ActionReject anAction, Mail aMail, ActionContext context) th
         }
 
         MailAddress soleRecipient = ActionUtils.getSoleRecipient(aMail);
-        String final_recipient = soleRecipient.toString();
-
+        String final_recipient = soleRecipient.asString();
         String original_message_id = aMail.getMessage().getMessageID();
 
-        DispositionModifier modifiers[] = {DispositionModifier.Error};
-        Disposition disposition = new Disposition(DispositionActionMode.Automatic,
-                DispositionSendingMode.Automatic, DispositionType.Deleted, modifiers);
-
-        MimeMultipart multiPart = MDNFactory.create(humanText.toString(),
-                reporting_UA_name, reporting_UA_product, original_recipient,
-                final_recipient, original_message_id, disposition);
+        MimeMultipart multiPart = MDNFactory.create(
```

Comment:
this call is too complex, the API is probably not good enough judging on simplicity criteria

Reply:
We should also expose API calls that do nt require to create the fields. Will do.

Reply:
Explained in your earlier (duplicated) comment

Reply:
Simplified